placementApp/serializers.py

```diff
@@ -60,8 +67,18 @@ class Meta:
 
 
 class PositionSerializer(serializers.ModelSerializer):
-    company = serializers.PrimaryKeyRelatedField(queryset=Company.objects.all())
+    #    company = serializers.PrimaryKeyRelatedField(queryset=Company.objects.all())
+    company = CompanySerializer()
 
     class Meta:
         model = Position
         fields = "__all__"
+
+
+class ApplicationSerializer(serializers.ModelSerializer):
+    position = serializers.PrimaryKeyRelatedField(queryset=Position.objects.all())
+    student = StudentSerializer(read_only=True)
```

Comment:
Not the correct approach. Requires unnecessary filling of data. Is student data actually need if he is signed in ?

Reply:
Have made the changes